### ISSUE TITLE
Explore cards: Optimized “disjoint set” with Path Compression and Union by Rank

### ISSUE THREAD

**Issue by andronov-alexey** [LeetCode-Feedback/LeetCode-Feedback#21159]
### LeetCode Username
aandronov

### Problem number, title, and link
Optimized “disjoint set” with Path Compression and Union by Rank
https://leetcode.com/explore/learn/card/graph/618/disjoint-set/3843/

### Category of the bug
- [ ] Editorial

### Description of the bug
There's no mention about why ranks aren't updated during Path Compression

### Code used for Submit/Run operation
```
int find(int x) {
    if (x == root[x]) {
        return x;
    }
    // ! some ranks may become obsolete
    return root[x] = find(root[x]);
}
```

### Expected behavior
It should be mentioned that obsolete ranks are not critical and that's why they are not updated in code examples

**Comment by LC-Pam** [comment]
Hello,

Thank you for reaching out to us.

Your reported issue has been promptly relayed to our dedicated team for thorough investigation. We appreciate your patience and understanding as we work to address and resolve this matter.

Rest assured, we will reach out to you as soon as we have the results and updates regarding the issue.

Once again, thank you for your support and cooperation. If you have any further questions or concerns in the meantime, please feel free to let us know.

Best regards,
LeetCode Support Team

**Comment by Chet-Sheng** [comment]
@andronov-alexey exactly! Interpreting rank as depth of tree in their sample code is just wrong.
Also worth noting this issue: https://github.com/LeetCode-Feedback/LeetCode-Feedback/issues/20623
With path compression, the maximal depth of tree is 3, and time complexity of `find` and `union` are both `O(1)`.
So union by rank is totally obsolute.

**Comment by andronov-alexey** [comment]
Thanks for the update, but the added comment is wrong:
```
# Some ranks may become obsolete so they are not updated
        self.root[x] = self.find(self.root[x])
        return self.root[x]
```
It should be something like this:
```
       # Each node along the path will point directly to new root and have a rank value of "2"
       # For the sake of simplicity we do not update ranks here
        self.root[x] = self.find(self.root[x])
        return self.root[x]
```
